### ISSUE TITLE
Outlook Add-in SSO example for Microsoft Graph

### ISSUE THREAD

**Issue by ksobolewskiVL** [OfficeDev/office-js#2524]
> Message from office-js bot:
We’re closing this issue because it has been inactive for a long time. We’re doing this to keep the issues list manageable and useful for everyone. If this issue is still relevant for you, please create a new issue.
> Thank you for your understanding and continued feedback.

 

# Provide required information needed to triage your issue 

I’m working with example generated by yo generator-office for SSO in Outlook Add-in, I have some questions regarding it and I am encountering some problems.  

## Your Environment 

 

* Platform: PC desktop, Office on the web

* Host: Outlook

* Office version number: -

* Operating System: Windows

* Browser (if using Office on the web): Chrome

## Content

1. Logging into Microsoft Graph when using multiple accounts/mailboxes in OWA. When I try to get access token and then fetch the data from Graph using Chrome wrong account to login is picked. In the dialog which appears I see a different login than the account I am logged into and I don’t see any option to change between Account A and Account B, there is only password input field to login. As I saw in code OfficeRuntime.auth.getAccessToken throws error 13012, so the API is not supported in this case and fallback dialog appers with wrong account login. On other browsers not using multiple accounts, the problem does not occur. How can I avoid this situation? 

2. When there is no possibility to use SSO as in the previous example, when the api was not available, fallback dialog appers. In the dialog I can select one of many of my accounts, but by accident it may be an incompatible account with the one I am logged into. Is there any best practice to solve problems like that?  

3. If there are multiple OfficeRuntime.auth.getAccessToken function calls at the same time, for example multiple click on get profile info for SSO example, there is no problem on first try, but on the second, addin starts to throw error 13013 and as well as some requests are not completed. Is this related to Microsoft Graph concurrent requests  limitations for Outlook? https://docs.microsoft.com/en-gb/graph/throttling#outlook-service-limits 

4. In the case of obtaining a token in OWA, we are always asked whether we agree to display a dialog box. Is there any solution to avoid this question? or is this required?

5. Is this connection to Microsoft Graph available for all licences or are there any restrictions? Can we determine if the current user can use Graph? https://docs.microsoft.com/en-us/javascript/api/requirement-sets?view=common-js-preview 

Thank you in advance

**Comment by exextoc** [comment]
Hey @ksobolewskiVL 

Can you share a short video regarding the issue (1) and (3)? 
For the rest of the queries, we'll escalate the same to the Microsoft Graph team.

**Comment by ksobolewskiVL** [comment]
Hi @exextoc,

Issue (1) It happened to me only once and I don't know if I will be able to reproduce it. If so, I will upload it later.

Issue (3) Video is here:
https://www.youtube.com/watch?v=3osqPpxm-1w
I have tried several examples of token retrieval and what I have observed. When multiple getAccessToken requests occurs at the same time or with a small interval, an auth error is displayed and not all requests are executed. By increasing the interval to 2000ms the blocking occurs less frequently. When retrieving a token with a longer interval, it looks like some requests are retrieved by sso and some by fallbackauthdialog (due to error 13013). On the other hand, obtaining a token once, I can do 10 requests to Graph and the error message do not appear and in REST queries, I have not observed limitations.

**Comment by exextoc** [comment]
Hi @ksobolewskiVL,
1. We are also not able to repro the scenario in 1. Please let us know if you see that happening again.
2. There is no recommendation for this case.
3. Do you have a use case where you need to make multiple getAccessToken call continuously? 
4. Can you please attach the image?
5. We are trying to get the details for this from the Microsoft Graph team.
